### DISCUSSION TITLE
Migrate from Mediawiki to Wiki.js

### DISCUSSION THREAD

**Discussion by seanatemerald** [requarks/wiki#3425] [Help / Questions]
**Question**
I'm looking for the best way to migrate from MediaWiki to your platform.  I've heard that there is a tool in development, and if so I would like to know a timeline for its release.

**Host Info (please complete the following information):**
OS: Centos 7
Wiki.js version: N/A
Database engine: mysql

**Comment by kommandantjut** [comment]
I personally used the WYSIWG editor and copy and pasted. Works decently if you have a small wiki. Thing is that you have to redo pictures and links.

**Comment by Morgul** [comment]
I'm currently in the middle of a migration from a home-grown wiki solution I'd written a few years ago (stored everything as markdown and json) to WikiJS. I can tell you what I've discovered, and how I've approached it. (Note: I know nothing about how the project owners intend people to migrate, this is me just armed with the docs and some development knowledge.)

First, it's helpful to decide what you need to migrate. I'd ask the following questions:
* Do I need page history?
* Do I need editor history?
* Do I need comments?

If, by chance, the answer is no to all of those, then the best thing to do is to render the current version of all the pages to markdown (via something like https://github.com/philipashlock/mediawiki-to-markdown or pandoc) and then save them to a folder in whatever hierarchy you want. Then, set up a blank wikijs and configure the local filesystem storage, point it at your folder, and "Import Everything".

From here, the decision tree's a bit more complex. If you don't mind all your pages being created by Administrator, with a timestamp of now, I'd use the GraphQL api and write a script to do the migration. (My node script for my use case came out to be about two hundred and sixty lines of code, so not trivial, but not big, either.)

If you absolutely MUST have everything (which I'm working on); the best option I've founds so far is to build a git repo like the filesystem option mentioned above, but commit for every revision of every page, setting the author to the author of the revision. This will get you everything, minus users and comments. Those you will still need to add via the GraphQL endpoints.

Hope that helps some.

**Comment by gagarine** [comment]
-  Do I need page history?  YES.
-  Do I need editor history? YES

Import from the [XML mediawiki export](https://www.mediawiki.org/wiki/Help:Export) would be great.
